lib/solarwinds_apm/inst/rack.rb

```diff
@@ -159,6 +162,9 @@ def sample(env, settings, options, profile_spans)
 
             SolarWindsAPM::API.log_start(:rack, report_kvs, env, settings)
 
+            # mark the trace as started for this request
+            env['SW_APM_TRACE_STARTED'] = 'true'
```

Comment:
If I understand correctly - this is similar to the way the async context is kept in node, and the question is, why wasn't it needed before? And if it wasn't what is it in NH that requires this feature?

Reply:
There are cases where rack is re-entered during the same request. In the past the context was kept and continued.
But if something breaks and context is around for a different reason, we will still continue it. This is a safer approach, because we know when to continue and when to clean up. 
It was needed long before ... cleaning up loose ends